drivers/sensor/adc_thermistor/adc_thermistor.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(adc_thermistor, CONFIG_SENSOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT zephyr_adc_thermistor
+
+struct adc_thermistor_config {
+	const struct adc_dt_spec adc;
+
+	const int32_t *lut;
+	size_t lut_size;
+};
+
+struct adc_thermistor_data {
+	struct k_mutex mutex;
+
+	int32_t val;
+};
+
+static int adc_thermistor_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct adc_thermistor_config *config = dev->config;
+	struct adc_thermistor_data *data = dev->data;
+	int ret;
+
+	uint16_t buf;
+	struct adc_sequence sequence = {
+		.buffer = &buf,
+		/* buffer size in bytes, not number of samples */
+		.buffer_size = sizeof(buf),
+	};
+
+	/* Allow fetching using ambient or die temperature channel */
+	switch (chan) {
+	case SENSOR_CHAN_ALL:
```

Comment:
Could you clarify if the real meaning here is ALL or ANY?

Reply:
`SENSOR_CHAN_ALL` is used in other drivers as well.

I was trying to come up with a simple solution for this driver to be used, as it can be for both `SENSOR_CHAN_DIE_TEMP` or `SENSOR_CHAN_AMBIENT_TEMP`.

Adding a device tree property for this felt like overkill.